sn_client/src/audit/dag_crawling.rs

```diff
@@ -382,47 +377,49 @@ impl Client {
                     }
                 }
                 let spends_len = spends.len();
-                debug!("Depth {depth} - Got {spends_len} spends for parent Tx: {parent_tx_hash:?}");
-                trace!("Spends for {parent_tx_hash:?} - {spends:?}");
-
-                // check if we reached the genesis Tx
-                known_txs.insert(parent_tx_hash);
-                if parent_tx == *sn_transfers::GENESIS_CASHNOTE_PARENT_TX
-                    && spends
-                        .iter()
-                        .all(|s| s.spend.unique_pubkey == *sn_transfers::GENESIS_SPEND_UNIQUE_KEY)
+                debug!("Depth {depth} - Got {spends_len} spends for parent: {addrs_to_verify:?}");
+                trace!("Spends for {addrs_to_verify:?} - {spends:?}");
+
+                // check if we reached the genesis spend
+                known_ancestors.extend(addrs_to_verify.clone());
+                if spends
+                    .iter()
+                    .all(|s| s.spend.unique_pubkey == *sn_transfers::GENESIS_SPEND_UNIQUE_KEY)
                     && spends.len() == 1
                 {
```

Comment:
Here we use to check that the Genesis we have was the original Genesis meaning:
- the unique pk is GENESIS
- and the TX/descendants it was spent to is correct too

By removing the second check, we will miss out on rogue Genesises: Genesis not spent to the determined descendant!